src/Voron/Impl/Journal/JournalWriter.cs

```diff
@@ -28,49 +28,87 @@ public unsafe class JournalWriter : IJournalWriter
         private SafeJournalHandle _readHandle = new SafeJournalHandle();
         private int _refs;
         private bool _workingSetQuotaLogged = false;
+        private readonly StorageEnvironmentSynchronization _writeSynchronization;
+        private readonly long _maxNumberOf4KbInSingleWrite;
 
         public int NumberOfAllocated4Kb { get; }
         public bool Disposed => _disposed.IsRaised();
         public VoronPathSetting FileName { get; }
         public bool DeleteOnClose { get; set; }
 
-        public JournalWriter(StorageEnvironmentOptions options, VoronPathSetting filename, long size, PalFlags.JournalMode mode = PalFlags.JournalMode.Safe)
+        public JournalWriter(StorageEnvironmentOptions options, VoronPathSetting filename, long size, StorageEnvironmentSynchronization writeSynchronization = null, PalFlags.JournalMode mode = PalFlags.JournalMode.Safe)
         {
-            _options = options;
+            _options = options;            
             FileName = filename;
             _log = LoggingSource.Instance.GetLogger<JournalWriter>(options.BasePath.FullPath);
 
+            if (writeSynchronization == null)
+                writeSynchronization = new StorageEnvironmentSynchronization();
+
+            // The write synchronization mechanism will require a max size and to be aligned to 4kb.  
+            _writeSynchronization = writeSynchronization;
+            _maxNumberOf4KbInSingleWrite = writeSynchronization.MaxWriteSize / 4096;
```

Comment:
```suggestion
            _maxNumberOf4KbInSingleWrite = writeSynchronization.MaxWriteSize /  (4L * Constants.Size.Kilobyte);
```

Let's keep this consistent.